tensorflow/training/docker/2.11/py3/Dockerfile.cpu

```diff
@@ -0,0 +1,319 @@
+
+RUN echo "hwloc_base_binding_policy = none" >> /usr/local/etc/openmpi-mca-params.conf \
+ && echo "rmaps_base_mapping_policy = slot" >> /usr/local/etc/openmpi-mca-params.conf
+
+ENV LD_LIBRARY_PATH=/usr/local/openmpi/lib:$LD_LIBRARY_PATH
+ENV PATH /usr/local/openmpi/bin/:$PATH
+
+# SSH login fix. Otherwise user is kicked off after login
+RUN sed 's@session\s*required\s*pam_loginuid.so@session optional pam_loginuid.so@g' -i /etc/pam.d/sshd
+
+# Create SSH key.
+RUN mkdir -p /root/.ssh/ \
+ && mkdir -p /var/run/sshd \
+ && ssh-keygen -q -t rsa -N '' -f /root/.ssh/id_rsa \
+ && cp /root/.ssh/id_rsa.pub /root/.ssh/authorized_keys \
+ && printf "Host *\n  StrictHostKeyChecking no\n" >> /root/.ssh/config
+
+WORKDIR /
+
+RUN apt-get update \
+ && apt-get install -y --no-install-recommends \
+    libbz2-dev \
+    libc6-dev \
+    libffi-dev \
+    libgdbm-dev \
+    liblzma-dev \
+    libncursesw5-dev \
+    libreadline-gplv2-dev \
+    libsqlite3-dev \
+    libssl-dev \
+    tk-dev \
+    ffmpeg \
+    libsm6 \
+    libxext6 \
+ && rm -rf /var/lib/apt/lists/* \
+ && apt-get clean
+
+RUN wget https://www.python.org/ftp/python/$PYTHON_VERSION/Python-$PYTHON_VERSION.tgz \
+ && tar -xvf Python-$PYTHON_VERSION.tgz \
+ && cd Python-$PYTHON_VERSION \
+ && ./configure \
+ && make -j $(nproc) \
+ && make install \
+ && rm -rf ../Python-$PYTHON_VERSION* \
+ # Starting from Python39, a xxx.pem file will be generated under /tmp folder during installation. Remove it to complete cleanup after installation from python source.
+ && rm /tmp/*.pem
+
+RUN ${PIP} --no-cache-dir install --upgrade \
+    pip \
+    setuptools
+
+# Some TF tools expect a "python" binary
+RUN ln -s $(which ${PYTHON}) /usr/local/bin/python \
+ && ln -s $(which ${PIP}) /usr/bin/pip
+
+RUN apt-get update && apt-get -y install protobuf-compiler
+
+# install PyYAML==5.1.2 to avoid conflict with latest awscli
+# # python-dateutil==2.8.0 to satisfy botocore associated with latest awscli
+RUN ${PIP} install --no-cache-dir -U \
+    pybind11 \
+    cmake==3.18.2.post1 \
+    scipy==1.8.0 \
+    "Pillow>=9.0.0<9.1.0" \
+    python-dateutil==2.8.1 \
+    "pyYAML>=5.4,<5.5" \
+    requests==2.27.1 \
+    "awscli<2" \
+    mpi4py==3.1.3 \
+    # Let's install TensorFlow separately in the end to avoid
+    # the library version to be overwritten
+ && ${PIP} install --no-cache-dir -U \
+    h5py==3.6.0 \
+    "absl-py>=0.9,<0.11" \
+    "opencv-python>=4.3,<5" \
+    "werkzeug>=2.0.2,<3" \
+    psutil==5.7.2 \
+    "urllib3>1.26.0<1.27.0" \
+    # The Following Issues Require the Upper Bound Set on Protobuf:
+    #    -  https://github.com/tensorflow/tensorflow/issues/53234  
+    #    -  https://github.com/protocolbuffers/protobuf/issues/9954
+    #    -  https://github.com/tensorflow/tensorflow/issues/56077
+    # Reference setup.py protobuf upper limit here:
+    #    - https://github.com/tensorflow/tensorflow/blob/master/tensorflow/tools/pip_package/setup.py#L105
+    "protobuf>=3.9.2,<3.20"
```

Comment:
Can we change this to something like `>=3.19,<3.20`, so that the window is not so wide? I believe new patch versions for protobuf are only being published for `3.16.*` and `>=3.19.*`.

Reply:
Will update! Good callout

Reply:
Updated